mgc/discrim/discrimOneSample.py

```diff
@@ -0,0 +1,146 @@
+from sklearn.metrics import euclidean_distances
+from ._utils import _CheckInputs
+import numpy as np
+import random
+from .base import DiscriminabilityTest
+from scipy._lib._util import MapWrapper
+
+
+class DiscrimOneSample(DiscriminabilityTest):
+    r"""
+     A class that performs a one-sample test for whether the discriminability 
+     differs from random chance, as described in [#1Dscr]_.
+     
+     Discriminability index is a measure of whether a data acquisition and 
+     preprocessing pipeline is more discriminable among different subjects.
+     The key insight is that each measurement of the same item should be more 
+     similar to other measurements of that item, as compared to measurements 
+     of any other item. 
+    
+    Parameters
+    ---------- 
+    is_dist : Boolean, optional (default: False)
+        whether `x` is a distance matrix or not.
+    remove_isolates : Boolean, optional (default: True)
+        whether remove the samples with single instance or not.
+
+    See Also
+    --------
+    DiscrimTwoSample : Two sample test for comparing the discriminability of two data
+    
+    Notes
+    -----
+    With :math:`D_X` as the sample discriminability of :math:`X`, one sample test verifies 
+    whether
+    
+     .. math::
+
+         H_0: D_X = D_0
+
+    and
+
+     .. math::
+
+         H_A: D_X > D_0 
+    
+    where :math:`D_0` is the discriminability that would be observed by random chance.
+
+    References
+    ----------
+    .. [#1Dscr] Eric W. Bridgeford, et al. "Optimal Decisions for Reference 
+                Pipelines and Datasets: Applications in Connectomics." Bioarxiv (2019).
+    """
+
+    def __init__(self, is_dist=False, remove_isolates=True):
+        # set is_distance to true if compute_distance is None
+        self.is_distance = is_dist
+        self.remove_isolates = remove_isolates
+        DiscriminabilityTest.__init__(self)
+
+    def _statistic(self, x, y):
+        """
+        Helper function that calculates the discriminability test statistics.
+        """
+        stat_ = super(DiscrimOneSample, self)._statistic(x, y)
```

Comment:
rename to stat

Reply:
https://github.com/neurodata/mgc/blob/2fd2f674298926ac65c4cf61f76d65b8fdd61545/mgc/discrim/discrim_one_samp.py#L60-L66